### PR TITLE
Add new contenttype ToDo list

### LINKED ISSUE #5825: ToDo's should be organized in lists: Backend
Details see #5789 

Nachgeladen für Sprint #2

### Todos:
- [x] FTI TodoList 
 - Schema: Title, 
 - ID: Standard verhalten
 - Addable_types: Todos
- [x] Permission `Add portal content`
- [x] REST API
 - [x] CRUD TodoList
  - Delete: Validierung beim Löschen, nur empty.
 - [x] Support Move
 - [x] Support Ordering TodoList
 - [x] Support Ordering (https://plonerestapi.readthedocs.io/en/latest/content.html#reordering-sub-resources)
 - [x] Ordering analog Favoriten umsetzen bzw. analysieren und festhalten.

### PR DESCRIPTION
**Pull request by phgross**
This PR adds the new content type ToDo list, which is used for group todos in together. Closes #5825

- No UI is provided, only used by REST API
- I did not add example requests for the CRUD operations and instead referred to the corresponding chapter. I think that's enough.
- By default the ordering of content is only allowed with modifyPortalContent Permission on the container. But because a workspace member should be able to update the order but not to edit the workspace, I implemented a separate PATCH service for workspaces wich checks for a different permission.

## Checkliste

_Zutreffendes soll angehakt stehengelassen werden._
- [x] Profil angepasst? Sind UpgradeSteps vorhanden/nötig?:
- [x] Sind UpgradeSteps `deferrable`, oder können gewisse Schritte des Upgrades konditional ausgeführt werden?
_Da nur Profil Änderungen nicht notwendig._
- [x] Gibt es neue Übersetzungen?
  - [x] Sind alle msg-Strings in Übersetzungen Unicode?
  - [x] Wird die richtige i18n-domain verwendet (Copy-Paste Fehler sind hier häufig)?
- [x] Changelog-Eintrag vorhanden/nötig?
- [x] Aktualisierung Dokumentation vorhanden/nötig?

### CODE REVIEW

File: opengever/workspace/behaviors/namefromtitle.py [hunk 1]
```diff
@@ -30,3 +30,16 @@ class WorkspaceFolderNameFromTitle(dossiernamefromtitle.DossierNameFromTitle):
...
     format = u'folder-%i'
+
+
+class IToDoListNameFromTitle(INameFromTitle):
+    """Behavior interface for ToDoListNameFromTitle.
+    """
+
+
+class ToDoListNameFromTitle(dossiernamefromtitle.DossierNameFromTitle):
+    """ The ID of a workspace folder should be 'folder-{sequence number}'.
```
Comment (njohner) [inline opengever/workspace/behaviors/namefromtitle.py]: Comment needs to be updated

File: opengever/workspace/tests/test_todolist.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+                     method='POST', headers=self.api_headers,
+                     data=json.dumps(data))
+
+        self.assertEqual(200, browser.status_code)
+        self.assertEqual(
+            ['Fix user login'],
+            [obj.title for obj in self.todolist_general.objectValues()])
+
+    @browsing
+    def test_move_todo_from_on_todolist_to_another(self, browser):
```
Comment (njohner) [inline opengever/workspace/tests/test_todolist.py]: `one` not `on`

File: docs/public/dev-manual/api/workspace/todolists.rst [hunk 3]
```diff
@@ -3,4 +3,4 @@
 ToDo Listen
 ===========
 
-ToDo Listen dienen als Gruppierungsmöglicheiten für ToDos. Sie können via REST Operationen :ref:`operations` erstellt, gelesen, bearbeitet und gelöscht werden.
+ToDo Listen dienen als Gruppierungsmöglicheiten für ToDos. Sie können via REST Operationen :ref:`operations` erstellt, gelesen, bearbeitet und gelöscht werden. Beim Löschen gilt es zu beachten das nur leere Listen gelöscht werden können.
```
Comment (njohner) [inline docs/public/dev-manual/api/workspace/todolists.rst]: typo: zu beachten das -> zu beachten, dass

File: opengever/workspace/subscribers.py [hunk 4]
```diff
@@ -9,3 +10,16 @@ def assign_owner_role_on_creation(workspace, event):
     owner_userid = workspace.Creator()
     assignment = SharingRoleAssignment(owner_userid, ['WorkspaceOwner'])
     RoleAssignmentManager(workspace).add_or_update_assignment(assignment)
+
+
+def check_delete_preconditions(todolist, event):
+    """Its only possible to remove todolist if there are empty
```
Comment (njohner) [inline opengever/workspace/subscribers.py]: typo: there are empty -> they are empty

File: opengever/workspace/subscribers.py [hunk 5]
```diff
@@ -9,3 +10,16 @@ def assign_owner_role_on_creation(workspace, event):
     owner_userid = workspace.Creator()
     assignment = SharingRoleAssignment(owner_userid, ['WorkspaceOwner'])
     RoleAssignmentManager(workspace).add_or_update_assignment(assignment)
+
+
+def check_delete_preconditions(todolist, event):
```
Comment (njohner) [inline opengever/workspace/subscribers.py]: As discussed, this also prevents things like deleting a workspace, but that is fine.

File: opengever/workspace/tests/test_todolist.py [hunk 6]
```diff
@@ -158,3 +158,33 @@ def test_change_order_via_api(self, browser):
             ['folder-1', u'todolist-2', u'todolist-3',
              'opengever-workspace.todo', u'todolist-1', u'todolist-4'],
             self.workspace.objectIds())
+
+    @browsing
+    def test_move_todo_in_to_a_todolist(self, browser):
```
Comment (njohner) [inline opengever/workspace/tests/test_todolist.py]: `in_to` -> `into`

File: opengever/workspace/subscribers.py (hunk 5 above)
Comment (phgross) [reply opengever/workspace/subscribers.py]: Yes, but for now thats correct imho.

**Pull request merged**